src/ArmTemplates/Extractor/Utilities/DataProcessors/GroupDataProcessor.cs

```diff
@@ -0,0 +1,64 @@
+﻿// --------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+// --------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Templates.Groups;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.Models;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.Utilities.DataProcessors.Absctraction;
+
+namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.Utilities.DataProcessors
+{
+    public class GroupDataProcessor: IGroupDataProcessor
+    {
+        public IDictionary<string, string> OverrideRules {
+            get => this.overrideRules;  
+            set => this.overrideRules = value;
+        }
+
+        IDictionary<string, string> overrideRules;
+
+        public GroupDataProcessor() {
+            this.OverrideRules = new Dictionary<string, string>() {
+                { "Administrators", "administrators" },
+                { "Developers", "developers" },
+                { "Guests", "guests" }
+            };
+        }
+
+        public void ProcessData(List<GroupTemplateResource> groupTemplates, ExtractorParameters extractorParameters) {
+            
```

Comment:
1) indent
2) we can stop processing of data immediately if either `groupsTemplates.IsNullOrEmpty() == true` or `extractorParameters.OverrideGroupNames == false`

Reply:
Agree, thank you! Added the validation to other data processors as well